src/khoj/database/adapters/__init__.py

```diff
@@ -1115,30 +1115,36 @@ async def aset_user_voice_model(user: KhojUser, model_id: str):
 
     @staticmethod
     def get_chat_model(user: KhojUser):
-        subscribed = is_user_subscribed(user)
-        if not subscribed:
-            return ConversationAdapters.get_default_chat_model(user)
         config = UserConversationConfig.objects.filter(user=user).first()
         if config:
             return config.setting
+        subscribed = is_user_subscribed(user)
+        if not subscribed:
+            return ConversationAdapters.get_default_chat_model(user)
```

Comment:
This check should come before the config check, in case the user is not subscribed & had an older setting that's using an upgraded chat model. We can also use this function to opportunistically correct that mismatch.